agent_application/build.rs

```diff
@@ -0,0 +1,42 @@
+use std::env;
+use std::fs::File;
+use std::io;
+use std::io::BufWriter;
+use std::io::Write;
+use std::path::Path;
+
+fn main() -> io::Result<()> {
+    let out_dir = env::var("OUT_DIR").unwrap();
+    let file = File::create(Path::new(&out_dir).join("metadata.rs"))?;
+
+    let mut writer = BufWriter::new(file);
+
+    let app_version = env::var("APP_VERSION").ok();
+    let git_commit_hash = env::var("GIT_COMMIT_HASH").ok();
+    let docker_build_timestamp = env::var("DOCKER_BUILD_TIMESTAMP").ok();
+    let app_release_channel = env::var("APP_RELEASE_CHANNEL").ok();
+
+    writeln!(writer, "// THIS FILE IS GENERATED BY A BUILD SCRIPT")?;
+    writeln!(writer, "pub const APP_VERSION: Option<&str> = {:?};", app_version)?;
+    writeln!(
+        writer,
+        "pub const GIT_COMMIT_HASH: Option<&str> = {:?};",
+        git_commit_hash
+    )?;
+    writeln!(writer, "#[allow(dead_code)]")?;
+    writeln!(
+        writer,
+        "pub const DOCKER_BUILD_TIMESTAMP: Option<&str> = {:?};",
+        docker_build_timestamp
+    )?;
+    writeln!(writer, "#[allow(dead_code)]")?;
+    writeln!(
+        writer,
+        "pub const APP_RELEASE_CHANNEL: Option<&str> = {:?};",
+        app_release_channel
+    )?;
```

Comment:
You could consider using literal raw string notation to make this (slightly) more readable like this:
```rust
    writeln!(
        writer,
        r#"// THIS FILE IS GENERATED BY A BUILD SCRIPT
pub const APP_VERSION: Option<&str> = {app_version:?};
pub const GIT_COMMIT_HASH: Option<&str> = {git_commit_hash:?};
#[allow(dead_code)]
pub const DOCKER_BUILD_TIMESTAMP: Option<&str> = {docker_build_timestamp:?};
#[allow(dead_code)]
pub const APP_RELEASE_CHANNEL: Option<&str> = {app_release_channel:?};
"#
    )?;
```